uxarray/grid.py

```diff
@@ -405,11 +406,13 @@ def build_face_edges_connectivity(self):
         first_fill_value_index = np.argmax(closed == -1, axis=1)
         first_node = mesh2_face_nodes[:, 0]
 
-        # Now replace the first -1 at each row to be the first node
-        pad_results = np.array(
-            list(
-                map(self.__replace_fill_val, closed, first_fill_value_index,
-                    first_node)))
+        # Create 1D index into a 2D array
+        first_fill_idx = (m + 1) * np.arange(0, n) + first_fill_value_index
+
+        # replace all values at 1D index with the first_node
+        np.put(closed.ravel(), first_fill_idx, first_node)
```

Comment:
small variable name changes & comments. might be worth adding some comments about the 1D indexing, but I think it's self explanatory 

```Python
first_node = mesh2_face_nodes[:, 0]

# first index along columns where a fill value is present
first_fill_idx = np.argmax(closed == -1, axis=1)

# above index into a flattened (1D) array 
first_fill_idx_1d = (m + 1) * np.arange(0, n) + first_fill_idx

# replace first fill value with the first node
np.put(closed.ravel(), first_fill_idx, first_node)